alkira/resource_alkira_peering_gateway_cxp.go

```diff
@@ -0,0 +1,173 @@
+package alkira
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/alkiranet/alkira-client-go/alkira"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceAlkiraPeeringGatewayCxp() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Manage CXP Peering Gateways.",
+		CreateContext: resourceAlkiraPeeringGatewayCxpCreate,
+		ReadContext:   resourceAlkiraPeeringGatewayCxpRead,
+		UpdateContext: resourceAlkiraPeeringGatewayCxpUpdate,
+		DeleteContext: resourceAlkiraPeeringGatewayCxpDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "The name of the Peering Gateway.\nOnce deployed this property can not be changed. All the changes will be ignored.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"description": {
+				Description: "Description of the Peering Gateway.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"cxp": {
+				Description: "The CXP to which the Gateway is attached.\nOnce deployed this property can not be changed. All the changes will be ignored.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+
+			// TODO: change cloud_provider to be a Required value when more cloud providers are added and remove the default value.
+			"cloud_provider": {
+				Description: "The cloud provider on which the gateway will be created.\nOnce deployed this property can not be changed. All the changes will be ignored.",
+				Type:        schema.TypeString,
+				// Required:    true,
+				Optional: true,
+				Default:  "AZURE",
+			},
+			"cloud_region": {
+				Description: "The cloud region on which the ATH will be created. Eg : eastus , westus.\nOnce deployed this property can not be changed. All the changes will be ignored.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"segment": {
+				Description: "The name of the segment in which the gateway is created.\nOnce deployed this property can not be changed. All the changes will be ignored.",
```

Comment:
We do use `segment_id` in TF. This should be consistent with all other resources. We will take `segment_id` and convert to segment name internally. Refer to other resources.

Reply:
done.